src/main/java/net/minecraftforge/fml/network/NetworkRegistry.java

```diff
@@ -116,6 +122,10 @@ public static EventNetworkChannel newEventChannel(final ResourceLocation name, S
      */
     private static NetworkInstance createInstance(ResourceLocation name, Supplier<String> networkProtocolVersion, Predicate<String> clientAcceptedVersions, Predicate<String> serverAcceptedVersions)
     {
+        if(lock && !name.getNamespace().equals("fml")){
```

Comment:
FML is naughty and registers it's channels when connecting to a server the first time / accepting a client the first time.
So it's exempt from the lock.